LibreHardwareMonitorLib/Hardware/Computer.cs

```diff
@@ -318,15 +302,34 @@ public void Traverse(IVisitor visitor)
             }
         }
 
+        private void HardwareAddedEvent(IHardware hardware)
+        {
+            HardwareAdded?.Invoke(hardware);
+        }
+
+        private void HardwareRemovedEvent(IHardware hardware)
+        {
+            HardwareRemoved?.Invoke(hardware);
+        }
+
         private void Add(IGroup group)
         {
+            if (group == null)
+                return;
+
             lock (_lock)
             {
                 if (_groups.Contains(group))
                     return;
 
-
                 _groups.Add(group);
+
+                if ((group as IHardwareChanged) != null)
```

Comment:
Can you change this and the block below to:

> if (group is IHardwareChanged hardwareChanged)
{
hardwareChanged.HardwareAdded += ... 
...
}

Reply:
Done